common/src/config.rs

```diff
@@ -22,6 +23,23 @@ pub struct Database {
     pub name: String,
 }
 
+// TODO: figure out how to make clap use this and remove the redundant
+// #[arg(...)]'s above
+impl Default for Database {
```

Comment:
I think something around that could work:

```rust
fn test() {
    use clap::Parser;

    #[derive(clap::Parser, Debug, Clone)]
    pub struct Cli {
        #[command(flatten)]
        database: Database,
    }

    let _ = Cli::try_parse().unwrap().database;
}
```

You could also use the `clap::Parser` derive directly on the `Database` struct. I don't know if that has any side-effects.

Reply:
What would the `#[arg(...)]` lines become in the `Database` struct itself?

Reply:
> What would the `#[arg(...)]` lines become in the `Database` struct itself?

They wouldn't change.

Reply:
Then I'd still have those defaults in two places? I'd like the `Default` impl to be the only place.

Reply:
Sorry, I don't understand what you're suggesting. I just want the defaults in a single place. And the only risk of panic is if `DB_PORT` isn't a number, which is probably fine.

Reply:
I was suggesting to replace the hand crafted `std::env::var` code to be replaced by the clap generated parser. Removing that from the `Default` trait into some `fn` which clearly states that "this can panic".

Reply:
Hmm. Relying on clap macros to specify default values for structs used in a non-CLI context seems wrong to me.

Reply:
> in a non-CLI 

yeah :+1:

Reply:
@ctron with this last change, I'm using `default_value_t = Self::default().username`. are you ok with that?